core/service.py

```diff
@@ -9,7 +9,7 @@ def __init__(self, name, connector_func, state_processor_method=None,
         self.state_processor_method = state_processor_method
         self.names_previous_services = names_previous_services or set()
         self.names_required_previous_services = names_required_previous_services or set()
-        self.tags = tags or []
+        self.tags = set(tags) or set()
```

Comment:
```suggestion
        self.tags = set(tags or [])
```
`set(tags)` will raise `TypeError` if `tags` is `None` (by default).